nano/core_test/network.cpp

```diff
@@ -15,43 +15,14 @@ using namespace std::chrono_literals;
 
 TEST (network, tcp_connection)
 {
-	boost::asio::io_context io_ctx;
-	boost::asio::ip::tcp::acceptor acceptor (io_ctx);
+	nano::system system;
 	auto port = nano::get_available_port ();
-	boost::asio::ip::tcp::endpoint endpoint (boost::asio::ip::address_v4::any (), port);
-	acceptor.open (endpoint.protocol ());
-	acceptor.set_option (boost::asio::ip::tcp::acceptor::reuse_address (true));
-	acceptor.bind (endpoint);
-	acceptor.listen ();
-	boost::asio::ip::tcp::socket incoming (io_ctx);
-	std::atomic<bool> done1 (false);
-	std::string message1;
-	acceptor.async_accept (incoming,
-	[&done1, &message1] (boost::system::error_code const & ec_a) {
-		   if (ec_a)
-		   {
-			   message1 = ec_a.message ();
-			   std::cerr << message1;
-		   }
-		   done1 = true; });
-	boost::asio::ip::tcp::socket connector (io_ctx);
-	std::atomic<bool> done2 (false);
-	std::string message2;
-	connector.async_connect (boost::asio::ip::tcp::endpoint (boost::asio::ip::address_v4::loopback (), acceptor.local_endpoint ().port ()),
-	[&done2, &message2] (boost::system::error_code const & ec_a) {
-		if (ec_a)
-		{
-			message2 = ec_a.message ();
-			std::cerr << message2;
-		}
-		done2 = true;
-	});
-	while (!done1 || !done2)
-	{
-		io_ctx.poll ();
-	}
-	ASSERT_EQ (0, message1.size ());
-	ASSERT_EQ (0, message2.size ());
+	nano::simple_server_socket server (system.io_ctx, boost::asio::ip::address_v4::any (), port);
+	nano::simple_client_socket client (system.io_ctx, boost::asio::ip::address_v4::loopback (), port);
+
+	ASSERT_TIMELY (5s, server.connected && client.connected);
+	ASSERT_EQ (0, client.error_message.size ());
+	ASSERT_EQ (0, server.error_message.size ());
```

Comment:
The behavior of this test has changed. The original test was opening a socket, connecting and then sending a message. The new test does not send a message. Is that on purpose? 
Just thinking out loud - the following does not belong to this pull request:  `simple_server_socket`  and `simple_client_socket` are only used in this test. It would be better if the test could use the real `socket` and `server_socket` classes, but I can see why that is hard right now: They have dependencies to `node`, `stat` and `node_config`, which prevent the simple creation of a socket.

Reply:
It did not send, the message are variable that receives the error code, one for accept, another for connect. So this is just testing a connection can happen in the test environment without the nano classes for it. I was thinking about using simple_socket classes as an alternative to the nano classes server_socket, so we just use the nano classes when in fact testing it. But it is open to any path we decide to follow.